src/CoreRCON/RCON.cs

```diff
@@ -367,39 +369,48 @@ public async Task<string> SendCommandAsync(string command, TimeSpan? overrideTim
         private void RCONPacketReceived(RCONPacket packet)
         {
             _logger?.LogTrace("RCON packet received: {}", packet.Id);
+            
+            TaskCompletionSource<string> taskSource;
             // Call pending result and remove from map
-            if (_pendingCommands.TryGetValue(packet.Id, out TaskCompletionSource<string> taskSource))
+            if (!_pendingCommands.TryGetValue(packet.Id, out taskSource))
             {
-                if (_multiPacket)
+                _logger?.LogWarning("Received packet with no matching command id: {} body: {}", packet.Id, packet.Body);
+                // The server did not respect our id
+                if (!strictCommandPacketIdMatching && packet.Id == 0)
                 {
-                    //Read any previous messages 
-                    _incomingBuffer.TryGetValue(packet.Id, out string body);
-
-                    if (packet.Body == Constants.MULTI_PACKET_END_RESPONSE)
-                    {
-                        //Avoid yielding
-                        taskSource.SetResult(body ?? string.Empty);
-                        _pendingCommands.TryRemove(packet.Id, out _);
-                        _incomingBuffer.Remove(packet.Id);
-                    }
-                    else
-                    {
-                        //Append to previous messages
-                        _incomingBuffer[packet.Id] = body + packet.Body;
-                    }
+                    // Get the most recently sent command
+                    taskSource = _pendingCommands
+                        .OrderBy(cmd => cmd.Key)
+                        .Select(cmd => cmd.Value)
+                        .FirstOrDefault();
```

Comment:
If the server doesn't support multiple packages we could just do first or default here.

Reply:
Updated